emu/src/arm7tdmi.rs

```diff
@@ -122,6 +122,64 @@ impl Arm7tdmi {
         }
     }
 
+    fn data_transfer_immediate_offset(&mut self, op_code: ArmModeOpcode) -> bool {
+        let pre_post = op_code.get_bit(24);
+        let up_down = op_code.get_bit(23);
+        let _write_back = op_code.get_bit(21);
+        let load_store = op_code.get_bit(19);
+        let rn_base_register = op_code.get_bits(16..=19);
+        let rd_source_destination_register = op_code.get_bits(12..=15);
+        let transfer_type = HalfwordTransferType::from(op_code.get_bits(5..=6) as u8);
+        let immediate_offset_high = op_code.get_bits(8..=11);
+        let immediate_offset_low = op_code.get_bits(0..=3);
+
+        let offset = (immediate_offset_high << 4) | immediate_offset_low;
+        let mut address = rn_base_register;
+
+        if rn_base_register == 0xF {
+            // prefetching
+            let v: u32 = self.registers.program_counter().try_into().unwrap();
+            address = address.wrapping_add(v + 8);
+        }
```

Comment:
When `rn != 0xF` we're not reading its value, I would do something like:

```rust
let mut address = self.registers.register_at(rn_base_register.try_into().unwrap());

if rn_base_register == 0xF {
    address = address.wrapping_add(address + 8);
}
```

Reply:
Fixed!